pkg/registry/extensions/replicaset/strategy_test.go

```diff
@@ -141,3 +144,71 @@ func TestReplicaSetStatusStrategy(t *testing.T) {
 		t.Errorf("Unexpected error %v", errs)
 	}
 }
+
+func TestSelectorImmutability(t *testing.T) {
+	tests := []struct {
+		url                    string
+		oldSelectorLabels      map[string]string
+		newSelectorLabels      map[string]string
+		expectedAPIGroup       string
+		expectedAPIVersion     string
+		expectedSelectorLabels map[string]string
+	}{
+		{"/apis/apps/v1beta2/namespaces/default/replicasets/test-replicaset", map[string]string{"a": "b"}, map[string]string{"c": "d"}, "apps", "v1beta2", map[string]string{"a": "b"}},
+		{"/apis/apps/v1/namespaces/default/replicasets/test-replicaset", map[string]string{"a": "b"}, map[string]string{"c": "d"}, "apps", "v1", map[string]string{"a": "b"}},
+		{"/apis/extensions/v1beta1/namespaces/default/replicasets/test-replicaset", map[string]string{"a": "b"}, map[string]string{"c": "d"}, "extensions", "v1beta1", map[string]string{"c": "d"}},
+	}
+
+	resolver := newTestRequestInfoResolver()
+
+	for _, test := range tests {
+		req, _ := http.NewRequest("PUT", test.url, nil)
+
+		apiRequestInfo, err := resolver.NewRequestInfo(req)
```

Comment:
I don't think you need to go all the way back to parsing request info from urls in this test... just build a request info directly to test the various api groups